ingestion/src/metadata/utils/helpers.py

```diff
@@ -61,11 +63,18 @@ def get_database_service_or_create(
     if service:
         return service
     else:
+        password = (
+            config.password.get_secret_value()
+            if isinstance(config.password, SecretStr)
+            else config.password
+            if hasattr(config, "password")
+            else None
+        )
         service = {
             "databaseConnection": {
                 "hostPort": config.host_port if hasattr(config, "host_port") else None,
                 "username": config.username if hasattr(config, "username") else None,
-                "password": config.password if hasattr(config, "password") else None,
+                "password": password,
```

Comment:
https://github.com/open-metadata/OpenMetadata/pull/2385/files#diff-ef5072bce80b525315f2c4f3b0bff7f6d49dd26afa44ef2815b2c816f9d14152R66-R72
Lines from 66 to 72 will ensure that if the password is empty or not configured, it will send a None